### ISSUE TITLE
Custom CSS not saved

### ISSUE THREAD

**Issue by heliostatic** [theCrius/tildes-extended#32]
When adding custom CSS, it seems to overfill the localStorage key:

`Unchecked runtime.lastError while running storage.set: QUOTA_BYTES_PER_ITEM quota exceeded`

## Expected Behavior
Custom CSS should be saved.

## Current Behavior
It is not.

## Steps to Reproduce
Add these rules to custom css and save, watch console errors:
```
.topic-listing .is-topic-official, .topic-listing .is-topic-official .topic-info {
    border-color: #FF5555 !important;
    background: rgb(68, 70, 89)!important;
}
body .post-button {
    border: 1px dashed #8be9fd !important;
    padding: 10px !important;
    margin-right: 10px !important;
}
p {max-width: 95%; text-align: justify;}
blockquote { max-width: 92%; }
#sidebar {min-width: 250px; width: 250px;}
body .post-button-used {color: #FF8C00; }
.btn-used { background-color: #FF8C00; color: #FFFFFF; }
.is-comment-mine > .comment-itself { border-left: 3px solid orangered !important; }
.is-comment-mine > .comment-itself > header { background: gold !important; }
.comment .is-comment-by-op {
    background: linear-gradient(-90deg, rgba(190, 147, 249,0), rgba(190, 147, 249,0.14));
}
body a[href^="/user/"]:not(.link-user):not(.logged-in-user-username) {color:#FF79C6!important}
.is-comment-mine > .comment-itself {
  border-left: 3px solid #FF4136 !important;
}
.is-comment-mine > .comment-itself > header > .link-user {
  color: #FFFFFF !important;
  background: #0074D9 !important;
  padding: 3px 6px 3px 6px;
  border-radius: 7px;
}```

## Environment
Chrome 67, Mac OS 10.13

## Additional notes
<!--- Any other information you consider might be useful -->

**Comment by Bauke** [comment]
I'm not able to reproduce it on both Chrome and Firefox. Are you including any CSS from links? Also, what version of Tildes Extended are you using? You can find it if you right click the Tildes Extended and click "Manage Extensions", the current latest is 0.8.9

You could try entering `localStorage.clear();` in the console, see if that fixes it.

**Comment by heliostatic** [comment]
Yes, I did have css from links, and removing it does seem to have helped.

However, now there's another issue?

I only added one rule: ```
body {
  font-family: serif
}```
Then loading the page:
`Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src”). Source: body {
  font-family: serif
}.`

**Comment by theCrius** [comment]
Hey, it's probably a problem related to the full CSS (external + your custom) being too big for the standard storage size of the final variable that contains all the css to be injected.

I'll work on it and see what can be done about that.

Can you provide the exact links to the external CSS as well? So I can try and apply it to my local copy.

**Comment by heliostatic** [comment]
https://gitlab.com/Bauke/styles/raw/master/css/tildes-dracula/tildes-dracula.user.css
and
https://gitlab.com/Bauke/styles/raw/master/css/tildes-dracula/tildes-dracula.css

Thanks!

**Comment by Bauke** [comment]
Oh, you should only be importing one of them. They're both the same files except the `*.user.css` has some extra metadata for Stylus.

**Comment by heliostatic** [comment]
Done, thanks.

On Firefox Nightly, I'm getting this error with just `https://gitlab.com/Bauke/styles/raw/master/css/tildes-dracula/tildes-dracula.css` added, and no custom rules:
```Content Security Policy: The page’s settings blocked the loading of a resource at self (“style-src”). Source: 

@-moz-document domain('tildes.net') {
....```

**Comment by Bauke** [comment]
Have you toggled the `security.csp.enable` setting in `about:config`? It should be set to "false"

**Comment by heliostatic** [comment]
I had not. Just did, restarted browser, and no more console error, but the Dracula theme is still not applied.

**Comment by theCrius** [comment]
Now that you've disabled CSP, go back to setting, disable the external CSS feature. Save.
That should clean up anything stored as external CSS.

Enable it back and fill the input field again with the URL. Save. Now refresh tildes and the theme should apply.

*crossing fingers*

**Comment by heliostatic** [comment]
No dice on Firefox or Chrome.

Is there a csp setting on Chrome I need to set to false?

**Comment by Bauke** [comment]
Uhm, no. There shouldn't be one, only in Firefox.

Is it still not working?

**Comment by heliostatic** [comment]
No, still not working.

**Comment by Bauke** [comment]
Can you confirm this is what your custom CSS settings look like?

![image](https://user-images.githubusercontent.com/19501722/41789280-49709c1a-763e-11e8-924f-add686c8fdb4.png)

Seems to work for me on both Firefox and Chrome. Have you tried reinstalling the extension?

**Comment by heliostatic** [comment]
Removed and reinstalled the addon.
Options:
<img width="1007" alt="screen shot 2018-06-22 at 1 09 54 pm" src="https://user-images.githubusercontent.com/70898/41789578-cadc2c86-761d-11e8-85eb-e56e761bf897.png">

Incognito window with Tildes Extended enabled:
<img width="1792" alt="screen shot 2018-06-22 at 1 10 49 pm" src="https://user-images.githubusercontent.com/70898/41789590-d6bed788-761d-11e8-8eb3-43c0c58db9e6.png">

**Comment by Bauke** [comment]
Can you take a look inside the `<head/>` of the page?

![image](https://user-images.githubusercontent.com/19501722/41790029-ccb3aa52-7640-11e8-94a7-b66dbaf2b8e9.png)

There should be one with an id of customStylesheet.

**Comment by heliostatic** [comment]
![image](https://user-images.githubusercontent.com/70898/41790293-1122f830-7620-11e8-9fea-f31556af138f.png)

**Comment by Bauke** [comment]
Alright, I figured it out. It's the `@-moz-document domain('...')` that I use to identify the URL to which to apply the style on. Apparently this doesn't work on Chrome, as it's [not a Gecko engine](http://forums.mozillazine.org/viewtopic.php?f=25&t=2659307).

I tested this by cutting out the `@-moz-...` and sure enough, it works. Find [the snippet here](https://gitlab.com/snippets/1727205/raw).

Since it's not a Tildes Extended I'll be closing this one and making one in [my styles repo](https://gitlab.com/Bauke/styles), you'll be able to find updates on it there. I'll figure out a way to generate Chrome/Firefox specific files that work this time.
